src/Storages/Kafka/StorageKafka.cpp

```diff
@@ -966,9 +966,12 @@ void registerStorageKafka(StorageFactory & factory)
         {
             throw Exception(ErrorCodes::BAD_ARGUMENTS, "kafka_poll_max_batch_size can not be lower than 1");
         }
-        if (args.columns.getOrdinary() != args.columns.getAll() || !args.columns.getWithDefaultExpression().empty())
+        // Filter Default column type and skip Materialized, Ephemeral and Alias types.
+        NamesAndTypesList columnsWithDefaultExpr = args.columns.getWithDefaultExpression().filter(args.columns.getOrdinary().getNames());
```

Comment:
`clang-tidy` will be unhappy:

```suggestion
        NamesAndTypesList columns_with_default_expr = args.columns.getWithDefaultExpression().filter(args.columns.getOrdinary().getNames());
```